src/Filters/Issue.jsx

```diff
@@ -88,14 +88,44 @@ export class Issue extends React.Component {
     }
   }
 
-  _onOptionSelected(obj) {
-    // TODO: Select the parent + children
-    console.log('Selected "', obj.key, '"')
+  _onOptionSelected(item) {
+    // Find this option in the list
+    let idx = -1
+    for (let i = 0; i < this.props.options.length && idx === -1; i++) {
+      if (this.props.options[i].key === item.key) {
+        idx = i
+      }
+    }
+    console.assert(idx !== -1)
+
+    // Build a list of all the keys
+    const values = [item.key]
+    this.props.options[idx]["sub_issue.raw"].buckets.forEach(sub => {
+      values.push(slugify(item.key, sub.key))
+    })
+
+    this.props.typeaheadSelect(values)
   }
 }
 
 export const mapStateToProps = state => {
-  const options = state.aggs.issue || []
+  // See if there are an active issue filters
+  const allIssues = state.query.issue || []
+  const selections = []
+
+  // Reduce the issues to the parent keys (and dedup)
+  allIssues.forEach(x => {
+    const idx = x.indexOf(SLUG_SEPARATOR)
+    const key = (idx !== -1) ? x.substr(0, idx) : x
+    if (selections.indexOf(key) === -1)  {
+      selections.push(key)
+    }
+  })
+
+  // Make a cloned, sorted version of the aggs
+  const options = sortSelThenCount(state.aggs.issue, selections)
```

Comment:
Maybe! 

I put it here because this implementation definitely requires the sort, where the reducer may have to serve many different implementations.

It's a bit of an intellectual exercise since there is only one implementation needed in the application. So it works here or the reducer.